ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -234,30 +172,35 @@ public HelixClusterManager(ClusterMapConfig clusterMapConfig, String instanceNam
       helixClusterManagerMetrics.initializePartitionMetrics();
       helixClusterManagerMetrics.initializeCapacityMetrics();
     }
-    localDatacenterId = dcToDcZkInfo.get(clusterMapConfig.clusterMapDatacenterName).dcZkInfo.getDcId();
+    localDatacenterId = dcToDcInfo.get(clusterMapConfig.clusterMapDatacenterName).dcZkInfo.getDcId();
     partitionSelectionHelper =
         new PartitionSelectionHelper(helixClusterManagerCallback, clusterMapConfig.clusterMapDatacenterName,
             clusterMapConfig.clustermapWritablePartitionMinReplicaCount);
     // register partition selection helper as a listener of cluster map changes.
-    dcToDcZkInfo.values()
-        .forEach(info -> info.clusterChangeHandler.registerClusterMapListener(partitionSelectionHelper));
+    registerClusterMapListener(partitionSelectionHelper);
```

Comment:
I feel like it would, otherwise there would be no way to make a put request to newly added partitions in a cloud datacenter since the partitionSelectionHelper will not be fed CloudServiceReplicas as new partitions get added? Please correct me if my understanding is incorrect.